### PR TITLE
Add dashboard showing problematic nodes

### PR DESCRIPTION
**Pull request by adenitiu**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:
This PR adds a new panel in the Nodes/WorkerPool Overview dashboard, where one could see nodes that are unhealthy / unschedulable for more than 5 mins, during a time window of 30 mins, ignoring actions taken by the ClusterAutoscaler or by a possible K8S upgrade activity.

**Which issue(s) this PR fixes**:
This should provide a hint whether there is a problem with nodes which we would normally not spot as node becoming `NotReady` are usually self-healed.

**Special notes for your reviewer**:
/invite @rickardsjp @istvanballok 

**Release note**:

```other operator
Add unhealthy nodes dashboard.
```

### CODE REVIEW

**Review by acumino — COMMENTED** [review]
A couple of points - 
- I think you want to add this dashboard for every shoot, in that case, add the dashboard inside the dashboard/shoot folder.
- Why a separate dashboard? Probably, it should be added inside the `Node Details` dashboard.

**Comment by adenitiu** [comment]
@acumino thanks for the review. 
I added the dashboard under the `seed` folder, because I would like to monitor mostly only our Seeds, not all the problematic nodes on each enduser shoot.  To be honest, am not sure if the location is the correct one, hence am open for suggestions. 
And to answer the 2nd point: it's because I did not find any Node related dashboard in the `seed` folder.

File: pkg/component/plutono/plutono_test.go [hunk 1]
```diff
@@ -581,7 +581,7 @@ status:
 					Expect(string(managedResourceSecret.Data["configmap__some-namespace__plutono-datasources-27f1a6c5.yaml"])).To(Equal(dataSourceConfigMapYAMLFor(values)))
 					plutonoDashboardsConfigMap, err := getDashboardConfigMaps(ctx, c, namespace, "plutono-dashboards-[^-]{8}")
 					Expect(err).ToNot(HaveOccurred())
-					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: plutonoDashboardsConfigMap.Name}, 22)
+					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: plutonoDashboardsConfigMap.Name}, 23)
```
Comment (acumino) [inline pkg/component/plutono/plutono_test.go]: This change is not needed any more.
```suggestion
					testDashboardConfigMap(ctx, c, types.NamespacedName{Namespace: namespace, Name: plutonoDashboardsConfigMap.Name}, 22)
```

File: pkg/component/plutono/dashboards/shoot/owners/worker/node-pool-dashboard.json [hunk 2]
```diff
@@ -820,6 +820,99 @@
         "alignLevel": null
       }
     },
+    {
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": null,
+      "description": "Shows the number of nodes having issues for more than 5 mins in the cluster over a period of 30mins",
```
Comment (acumino) [inline pkg/component/plutono/dashboards/shoot/owners/worker/node-pool-dashboard.json]:
```suggestion
      "description": "Shows the number of nodes having issues for more than 5 mins in the cluster over a period of 30 mins.",
```

File: pkg/component/plutono/dashboards/shoot/owners/worker/node-pool-dashboard.json [hunk 3]
```diff
@@ -820,6 +820,99 @@
...
+          "exemplar": true,
+          "expr": "count_over_time((sum by (node) (kube_node_spec_taint{effect=\"NoSchedule\", key!=\"ToBeDeletedByClusterAutoscaler\", key!=\"node.gardener.cloud/critical-components-not-ready\"}))[30m:]) > 5",
+          "interval": "",
+          "legendFormat": "",
+          "refId": "A"
+        }
+      ],
+      "thresholds": [],
+      "timeRegions": [],
+      "title": "Nodes marked unhealthy, ignoring CA or K8S / GL upgrade activities",
```
Comment (acumino) [inline pkg/component/plutono/dashboards/shoot/owners/worker/node-pool-dashboard.json]: I am guessing, GL refers to Garden Linux, in that case.
```suggestion
      "title": "Nodes marked unhealthy, ignoring CA or K8S / OS upgrade activities.",
```

**Review by acumino — CHANGES_REQUESTED** [review]
It would be nice if you could post updated dashboard picture in the PR description. To have an idea of how the overall dashboard will look.

**Review by acumino — COMMENTED** [review]
Thanks!
/approve
/cc @istvanballok 
For dashboard query.

**Comment by adenitiu** [comment]
The dashboard would look like this
Screenshot 2023-12-05 at 11 41 32

**Comment by acumino** [comment]
> The dashboard would look like this

Not necessary, but the single dashboard should be aligned to the left. WDYT?

**Review by istvanballok — APPROVED** [review]
/lgtm
I double checked the promql expression and also tested the dashboard panel itself.

**Pull request merged**